### PR TITLE
✨(artifact): Fix formatArtifactToMarkdown output format for DML operations

### PR DESCRIPTION
**Pull request by MH4GF**
## Issue

- resolve: #3049

## Why is this change needed?

The formatArtifactToMarkdown utility was generating incorrect output format for DML operations. The function was wrapping operations in unnecessary headers and numbering that made the output less readable and not aligned with the expected format.

## Summary

- Updated `formatArtifactToMarkdown` implementation to format DML operations directly as level 5 headings
- Removed the separate `formatDmlOperation` function that is no longer needed
- Each operation is now formatted as `##### **OPERATION_TYPE** - Description` for better hierarchy
- Removed "Related DML Operations:" header and "Operation N" numbering for cleaner output
- Added comprehensive tests with realistic airline system example data
- Updated test expectations to match the new simplified format

## Test plan

- [x] All existing tests pass
- [x] Added comprehensive test with realistic data (airline system example)
- [x] Tests cover single and multiple DML operations
- [x] Tests verify correct formatting of execution logs (success/failure)
- [x] Snapshot tests updated to match new format

## Summary by CodeRabbit

- Refactor
  - Streamlined DML operation display with direct operation headings, immediate SQL blocks, and per-operation execution history.
  - Standardized execution timestamps to UTC and added success/failure icons.
  - Improved section behavior: correct numbering, and omission of empty functional/non-functional sections.

- Tests
  - Added comprehensive suite validating full artifact rendering, edge cases, SQL whitespace trimming, special characters, and execution log scenarios to ensure consistent Markdown output.

### CODE REVIEW

File: frontend/apps/app/components/SessionDetailPage/components/Output/components/Artifact/utils/__tests__/formatArtifactToMarkdown.test.ts [hunk 1]
```diff
@@ -0,0 +1,1145 @@
...
+              description:
+                'Pilot personal information must be encrypted for storage and access logs must be recorded. Database access must be restricted to authenticated users only.',
+              type: 'non_functional',
+            },
+          ],
+        },
+      }
+
+      const result = formatArtifactToMarkdown(artifact)
+      expect(result).toMatchInlineSnapshot(`
```
Comment (MH4GF) [inline frontend/apps/app/components/SessionDetailPage/components/Output/components/Artifact/utils/__tests__/formatArtifactToMarkdown.test.ts:229]: I'd like to see a review of this output to see if it is easy to understand.

File: frontend/apps/app/components/SessionDetailPage/components/Output/components/Artifact/utils/__tests__/formatArtifactToMarkdown.test.ts [hunk 2]
```diff
@@ -0,0 +1,1151 @@
...
+        INSERT INTO airplanes (airplane_number, model, capacity) VALUES
+          ('JA100A', 'ATR42-600', 48),
+          ('JA200B', 'Bombardier CRJ900', 90),
+          ('JA330C', 'Flying Car Alpha', 5);
+        COMMIT;
+        \`\`\`
+
+        **Execution History:**
+
+        ✅ **06/01/2024, 08:00:00 AM**
```
Comment (FunamaYukina) [inline frontend/apps/app/components/SessionDetailPage/components/Output/components/Artifact/utils/__tests__/formatArtifactToMarkdown.test.ts:266]: While ``June 1, 2024, 8:05 AM`` might feel more natural in English-speaking regions than ``06/01/2024, 08:00:00 AM,`` I think it’s perfectly fine to keep it as it is for now.

**Review by FunamaYukina — APPROVED** [review]
Thank you for adding the test!
LGTM!

File: frontend/apps/app/components/SessionDetailPage/components/Output/components/Artifact/utils/__tests__/formatArtifactToMarkdown.test.ts (hunk 2 above)
Comment (MH4GF) [reply frontend/apps/app/components/SessionDetailPage/components/Output/components/Artifact/utils/__tests__/formatArtifactToMarkdown.test.ts:266]: Thanks, I'd like to follow up and make some adjustments here 👍🏻
